src/providers/ad/ad_gpo.c

```diff
@@ -864,19 +864,20 @@ ad_gpo_filter_gpos_by_dacl(TALLOC_CTX *mem_ctx,
 
         access_allowed = false;
         candidate_gpo = candidate_gpos[i];
-        sd = candidate_gpo->gpo_sd;
-        dacl = candidate_gpo->gpo_sd->dacl;
```

Comment:
It had to be moved because the candidate_gpo->gpo_sd is NULL if the gPCFunctionalityVersion is missing, so the check for the value of that attribute should be performed sooner.